drivers/wasapi/audio_driver_wasapi.cpp

```diff
@@ -455,6 +455,7 @@ Error AudioDriverWASAPI::audio_device_init(AudioDeviceWASAPI *p_device, bool p_i
 Error AudioDriverWASAPI::init_output_device(bool p_reinit) {
 	Error err = audio_device_init(&audio_output, false, p_reinit);
 	if (err != OK) {
+		finish_output_device();
```

Comment:
Put it here and not in `audio_device_init` to not duplicate `audio_device_finish();` line every time there's an error in `audio_device_init`. `audio_device_init` is only called from `init_[input,output]_device` functions (and from itself too, but that's not as important), so `finish_output_device` will be called every time initialization fails